src/plugins/intel_cpu/src/nodes/common/cpu_convert.cpp

```diff
@@ -464,14 +599,141 @@ struct ConvertPrecision<std::tuple<ov::float16, ov::float16>> {
                 const size_t current_batch_size = std::min(ctx.size - offset, batch);
                 jit_convert(src + offset, tmp, current_batch_size);     // fp16 -> fp32
                 for (size_t j = 0; j < current_batch_size; ++j)         // truncate fp32
-                    tmp[j] = std::trunc(std::max(std::min(tmp[j], ubound), lbound));
+                    tmp[j] = clamp ? std::trunc(std::max(std::min(tmp[j], ubound), lbound)) : tmp[j];
                 jit_convert(tmp, dst + offset, current_batch_size);     // fp32 -> fp16
             });
         }
 
         ctx.converted = true;
     }
 };
+
+template <>
+struct ConvertPrecision<std::tuple<float, ov::float16>> {
+    void operator()(ConvertContext& ctx) {
+        auto src = static_cast<const float*>(ctx.srcPtr);
+        auto dst = static_cast<ov::float16*>(ctx.dstPtr);
+
+        constexpr size_t batch = 64;
+        const size_t iterations = ov::intel_cpu::div_up(ctx.size, batch);
+        auto clamp = ctx.clamp;
+
+        if (!clamp) {
+            parallel_for(iterations, [&](size_t i) {
+                const size_t offset = i * batch;
+                const size_t current_batch_size = std::min(ctx.size - offset, batch);
+                jit_convert(src + offset, dst + offset, current_batch_size);  // fp32 -> fp16
+            });
+        } else {
+            parallel_for(iterations, [&](size_t i) {
+                const size_t offset = i * batch;
+                const size_t current_batch_size = std::min(ctx.size - offset, batch);
+                jit_convert<float, ov::float16, true>(src + offset, dst + offset, current_batch_size);  // fp32 -> fp16
+            });
+        }
+
+        ctx.converted = true;
+    }
+};
+
+template <>
+struct ConvertPrecision<std::tuple<uint8_t, ov::float16>> {
+    void operator()(ConvertContext& ctx) {
+        auto src = static_cast<const uint8_t*>(ctx.srcPtr);
+        auto dst = static_cast<ov::float16*>(ctx.dstPtr);
+
+        constexpr size_t batch = 64;
+        const size_t iterations = ov::intel_cpu::div_up(ctx.size, batch);
+        auto clamp = ctx.clamp;
+
+        if (!clamp) {
+            parallel_for(iterations, [&](size_t i) {
+                const size_t offset = i * batch;
+                const size_t current_batch_size = std::min(ctx.size - offset, batch);
+                jit_convert(src + offset, dst + offset, current_batch_size);  // src_t -> fp16
+            });
+
+        } else {
+            typedef float batch_type[batch];
+            uint8_t lbound, ubound;
+            std::tie(lbound, ubound) = ctx.range<uint8_t>();
+            parallel_for(iterations, [&](size_t i) {
+                batch_type tmp;
+                const size_t offset = i * batch;
+                const size_t current_batch_size = std::min(ctx.size - offset, batch);
+                for (size_t j = 0; j < current_batch_size; ++j)  // src_t -> fp32
```

Comment:
Could you please elaborate why we can't use a jit function to convert precision from int8 to fp16 directly? So as for the counterparts in `ConvertPrecision<std::tuple<float, int8_t>>` and `ConvertPrecision<std::tuple<ov::float16, int8_t>>`.

Reply:
as synced offline, currently this PR only add  parts of all supported conversion types' JIT Optimization(align with ngraph impl ) and try to keep the rest conversion types' impl as much the same as previous cpu_convert impl.  Just as you mentioned some of the left types could be further optimized by JIT, but Maybe we could carry on these other types' optimization when there are specific request models/cases.